third_party/amd/include/TritonAMDGPUTransforms/Passes.td

```diff
@@ -124,4 +124,19 @@ def TritonAMDGPUConvertToBufferOps : Pass<"tritonamdgpu-convert-buffer-ops", "ml
   let dependentDialects = ["mlir::triton::amdgpu::TritonAMDGPUDialect"];
 }
 
+def TritonAMDGPUInThreadTranspose: Pass<"tritonamdgpu-in-thread-transpose", "mlir::ModuleOp"> {
+  let summary = "Transpose K-outer dot operand while data is loaded into register right before writing to LDS";
+
+  let description = [{
+    Transpose K-outer dot operand (not consecutive on K dimension) right before writing data into LDS. This feature
+    happens right after data has been loaded from global memory to register-local registers. This feature will
+    allow coalesced global loading while still maintaining vectorized LDS write/read, by adding few VALU instructions
```

Comment:
We guarantee vectorized LDS read, as required by MFMA instructions. There is no guarantee of vectorized LDS write.

Reply:
reworded this section to make more clear